api/proto/teleport/loginrule/v1/loginrule_service.proto

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.loginrule.v1;
+
+import "google/protobuf/empty.proto";
+import "teleport/loginrule/v1/loginrule.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/loginrule/v1";
+
+// LoginRuleService provides CRUD methods for the LoginRule resource.
+service LoginRuleService {
+  // CreateLoginRule creates a login rule if one with the same name does not
+  // already exist, else it returns an error.
+  rpc CreateLoginRule(CreateLoginRuleRequest) returns (LoginRule);
```

Comment:
Not released to this PR but the LoginRule proto def uses default json tags https://github.com/gravitational/teleport/blob/016ae265e6f4bf11f5276a8942cf7c7b08f40de5/api/proto/teleport/loginrule/v1/loginrule.proto#L26 

Do we decided to user and store the kind as json object in our backend without keeping snake case formatting ?

Reply:
Since this file uses protoc-gen-go instead of protoc-gen-gogofast which is used in the "legacy" protos, I think there is no way to explicitly set the json tags. But if you check the generated code, actually it gets the snake_case json tags by default since the fields in the proto file use snake_case. https://github.com/gravitational/teleport/blob/016ae265e6f4bf11f5276a8942cf7c7b08f40de5/api/gen/proto/go/teleport/loginrule/v1/loginrule.pb.go#L59

I am using the Metadata and StringValues types from the legacy proto package, I could duplicate them outside of "legacy" but I'm not sure if there's a reason tbh